addons/mail/models/discuss/discuss_channel.py

```diff
@@ -197,37 +197,42 @@ def _compute_invitation_url(self):
 
     @api.model_create_multi
     def create(self, vals_list):
+        channel_members = []
         for vals in vals_list:
             # find partners to add from partner_ids
-            partner_ids_cmd = vals.get('channel_partner_ids') or []
+            partner_ids_cmd = vals.pop('channel_partner_ids', [])
             if any(cmd[0] not in (4, 6) for cmd in partner_ids_cmd):
                 raise ValidationError(_('Invalid value when creating a channel with members, only 4 or 6 are allowed.'))
             partner_ids = [cmd[1] for cmd in partner_ids_cmd if cmd[0] == 4]
             partner_ids += [cmd[2] for cmd in partner_ids_cmd if cmd[0] == 6]
 
             # find partners to add from channel_member_ids
-            membership_ids_cmd = vals.get('channel_member_ids', [])
+            membership_ids_cmd = vals.pop('channel_member_ids', [])
             if any(cmd[0] != 0 for cmd in membership_ids_cmd):
                 raise ValidationError(_('Invalid value when creating a channel with memberships, only 0 is allowed.'))
-            membership_pids = [cmd[2]['partner_id'] for cmd in membership_ids_cmd if cmd[0] == 0]
+            members = [{
+                k: v
+                for k, v in cmd[2].items()
+                if k in ['partner_id', 'is_pinned']
```

Comment:
`guest_id` is also given (`test_upload_denied` for example, but it can also be done directly from the form view)
Maybe we should at least raise an error message if invalid keys are given, rather than silently ignoring them.

Reply:
I hesitated myself between both.